t/TestDNS.pm

```diff
@@ -91,13 +94,37 @@ sub gen_dns_reply ($$) {
         $answers = [$answers];
     }
 
+    my $authority_answers = $t->{authority} // [];
+    if (!ref $authority_answers) {
+	$authority_answers = [$authority_answers];	
```

Comment:
Style: okay, it's not 3-space indentation, but a tab. Please replace all use of tabs in your patch to 4 spaces. Thank you.